### ISSUE TITLE
cli: `cdk migrate` outputs an incorrect template for Lambda environment variables

### ISSUE THREAD

**Issue by ymhiroki** [aws/aws-cdk#28997]
### Describe the bug

I ran `cdk migrate` command with the CFn template below, which copied from [the blog post](https://aws.amazon.com/jp/blogs/devops/announcing-cdk-migrate-a-single-command-to-migrate-to-the-aws-cdk/).

```
AWSTemplateFormatVersion: "2010-09-09"
Description: AWS CDK Migrate Demo Template
Parameters:
  FunctionResponse:
    Description: Response message from the Lambda function
    Type: String
    Default: Hello World
  BucketTag:
    Description: The tag value of the S3 bucket
    Type: String
    Default: ChangeMe
Resources:
  LambdaExecutionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
            Action: sts:AssumeRole
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
  HelloWorldFunction:
    Type: AWS::Lambda::Function
    Properties:
      Role: !GetAtt LambdaExecutionRole.Arn
      Code:
        ZipFile: |
          import os
          def lambda_handler(event, context):
            function_response = os.getenv('FUNCTION_RESPONSE')
            return {
              "statusCode": 200,
              "body": function_response
            }
      Handler: index.lambda_handler
      Runtime: python3.11
      Environment:
        Variables:
          FUNCTION_RESPONSE: !Ref FunctionResponse
  S3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      PublicAccessBlockConfiguration:
        BlockPublicAcls: true
        BlockPublicPolicy: true
        IgnorePublicAcls: true
        RestrictPublicBuckets: true
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      Tags:
        - Key: Application
          Value: Git-Sync-Demo
        - Key: DynamicTag
          Value: !Ref BucketTag
Outputs:
  S3BucketName:
    Description: The name of the S3 bucket
    Value: !Ref S3Bucket
    Export:
      Name: !Sub ${AWS::StackName}-S3BucketName
```

The command generates the typescript code below for the Lambda function.

```
    const helloWorldFunction = new lambda.CfnFunction(this, 'HelloWorldFunction', {
      role: lambdaExecutionRole.attrArn,
      code: {
        zipFile: 'import os\ndef lambda_handler(event, context):\n  function_response = os.getenv(\'FUNCTION_RESPONSE\')\n  return {\n    \"statusCode\": 200,\n    \"body\": function_response\n  }\n',
      },
      handler: 'index.lambda_handler',
      runtime: 'python3.11',
      environment: {
        variables: {
          functionResponse: props.functionResponse!,
        },
      },
    });
```

`variables` should contain `FUNCTION_RESPONSE`, however, it's actually `functionResponse`.

### Expected Behavior

Lambda environment variables must contain `FUNCTION_RESPONSE` like the code below.

```
    const helloWorldFunction = new lambda.CfnFunction(this, 'HelloWorldFunction', {
      role: lambdaExecutionRole.attrArn,
      code: {
        zipFile: 'import os\ndef lambda_handler(event, context):\n  function_response = os.getenv(\'FUNCTION_RESPONSE\')\n  return {\n    \"statusCode\": 200,\n    \"body\": function_response\n  }\n',
      },
      handler: 'index.lambda_handler',
      runtime: 'python3.11',
      environment: {
        variables: {
          FUNCTION_RESPONSE: props.functionResponse!,
        },
      },
    });
```

### Current Behavior

`cdk migrate` command generates an incorrect template for Lambda environment variables.
The function doesn't work well since it can't read environment variables.

```
    const helloWorldFunction = new lambda.CfnFunction(this, 'HelloWorldFunction', {
      role: lambdaExecutionRole.attrArn,
      code: {
        zipFile: 'import os\ndef lambda_handler(event, context):\n  function_response = os.getenv(\'FUNCTION_RESPONSE\')\n  return {\n    \"statusCode\": 200,\n    \"body\": function_response\n  }\n',
      },
      handler: 'index.lambda_handler',
      runtime: 'python3.11',
      environment: {
        variables: {
          functionResponse: props.functionResponse!,
        },
      },
    });
```

### Reproduction Steps

Run `cdk migrate --stack-name sample-stack --language typescript --from-path ./demo-template.yaml`

with `demo-template.yaml`:
```
AWSTemplateFormatVersion: "2010-09-09"
Description: AWS CDK Migrate Demo Template
Parameters:
  FunctionResponse:
    Description: Response message from the Lambda function
    Type: String
    Default: Hello World
  BucketTag:
    Description: The tag value of the S3 bucket
    Type: String
    Default: ChangeMe
Resources:
  LambdaExecutionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
            Action: sts:AssumeRole
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
  HelloWorldFunction:
    Type: AWS::Lambda::Function
    Properties:
      Role: !GetAtt LambdaExecutionRole.Arn
      Code:
        ZipFile: |
          import os
          def lambda_handler(event, context):
            function_response = os.getenv('FUNCTION_RESPONSE')
            return {
              "statusCode": 200,
              "body": function_response
            }
      Handler: index.lambda_handler
      Runtime: python3.11
      Environment:
        Variables:
          FUNCTION_RESPONSE: !Ref FunctionResponse
  S3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      PublicAccessBlockConfiguration:
        BlockPublicAcls: true
        BlockPublicPolicy: true
        IgnorePublicAcls: true
        RestrictPublicBuckets: true
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      Tags:
        - Key: Application
          Value: Git-Sync-Demo
        - Key: DynamicTag
          Value: !Ref BucketTag
Outputs:
  S3BucketName:
    Description: The name of the S3 bucket
    Value: !Ref S3Bucket
    Export:
      Name: !Sub ${AWS::StackName}-S3BucketName
```

### CDK CLI Version

2.126.0 (build fb74c41)

### Node.js Version

v20.10.0

### OS

macOS Sonoma 14.3 (23D56)

### Language

TypeScript

**Comment by tim-finnigan** [comment]
Thanks for reporting this issue, this does appear to be a bug with the environment variable getting converted to camel case.

**Comment by TheRealAmazonKendra** [comment]
Duplicate of https://github.com/aws/aws-cdk/issues/29027

**Comment by TheRealAmazonKendra** [comment]
This fix for this was released in v2.132.
